numpy/_core/src/multiarray/stringdtype/casts.cpp

```diff
@@ -658,6 +852,135 @@ pyobj_to_string(PyObject *obj, char *out, npy_string_allocator *allocator)
     return 0;
 }
 
+template<NPY_TYPES typenum>
+static NPY_CASTING string_to_int_resolve_descriptors(
+    PyObject *NPY_UNUSED(self),
+    PyArray_DTypeMeta *NPY_UNUSED(dtypes[2]),
+    PyArray_Descr *given_descrs[2],
+    PyArray_Descr *loop_descrs[2],
+    npy_intp *NPY_UNUSED(view_offset)
+) {
+    if (given_descrs[1] == NULL) {
+        loop_descrs[1] = PyArray_DescrNewFromType(typenum);
+    }
+    else {
+        Py_INCREF(given_descrs[1]);
+        loop_descrs[1] = given_descrs[1];
+    }
+
+    Py_INCREF(given_descrs[0]);
+    loop_descrs[0] = given_descrs[0];
+
+    return NPY_UNSAFE_CASTING;
+}
+
+// Example template parameters:
+// TNpyType: npy_int8
+// TNpyLongType: npy_longlong
```

Comment:
same here, I also find these type names confusing

Reply:
Understood - I'll just use `NpyType` and `NpyLongType`, if that works?